### DISCUSSION TITLE
CDN URL is not prepended for /public directory when img src is dynamic

### DISCUSSION THREAD

**Discussion by preetmishra** [nuxt/nuxt#22461] [Questions]
When I put in a static URL in the `src`, the image is rendered on my local server and in production (with the CDN URL prefix as `https://cdn.company.com/images/banner.jpg`).

```vue
<img src="/images/banner.jpg" />
```

However, when the `src` is dynamic, the image only works on the local server, on production, the CDN URL is never prepended.

```vue
<script>
  const event = { banner: "/images/banner.jpg" }
</script>
<template>
  <img :src="event.banner" />
</template>
```

I would highly appreciate any pointers. Please let me know if any additional information is required. Thank you.

**Comment by deeluxe74** [comment]
Hi,
I got exactly the same problem, if you inspect the dist folder you can see that the image is not added when the src is bind.
You can solve this by importing the image manually like this:

`import myImage from '/assets/images/...'`
But it seems a really bad practice when using dynamic image, you can see full of it [ir](https://stackoverflow.com/questions/74688433/why-loading-dynamically-assets-fails-on-nuxt-v3).

I found a best way for importing on build, but the path is broken on prod, i try to rename the path on build with nitro, but without success, and it seems pretty bad...

The last and easiest solution for me was to add dynamic image inside the public folder. This works fine, and no need tricky update.
Is maybe not a good practice to use public for that, but after build it seems dist folder is a shortcut from public.

**Reply by preetmishra** [reply]
Hi, @deeluxe74. Thanks for your reply. To clarify, I have all my images in the `/public` directly already. They still don't work.

**Reply by deeluxe74** [reply]
You need to find the "why" the images doesn't work in prod.
For me, you have only two option:
- The path is not good (inspect the src inside the HTML renderer after build)
- The image is not available inside the dist folder (If the images is on `public/images/banner.jpg`, inside the dist you normaly have `dist/images/banner.jpg`.

Try to clarify this, and with that we can find a solution.

**Reply by preetmishra** [reply]
Thanks, @deeluxe74. The images don't work because the path is not right (it doesn't have my CDN URL prepended). If I manually do the following, it works. However, ideally, it should work without it.

```vue
<script>
  const event = { banner: "/images/banner.jpg" }
  const CDN_URL = env === "production" ? useRuntimeConfig().public.CDN_URL : ""
</script>
<template>
  <img :src="`${CDN_URL}${event.banner}`" />
</template>
```

**Reply by deeluxe74** [reply]
What is the value of `useRuntimeConfig().public.CDN_URL`?

I don't understand why you need this. Normaly in dev the path for your src is `public/images/banner.jpg`, and in prod `images/banner.jpg` (inside dist folder).

**Reply by preetmishra** [reply]
Apologies for the late reply. In production, the value is `https://cdn.company.com/images/banner.jpg`. The value is an empty string `''` locally. If I don't do this, my images have 404.

Ideally, since I've put the `CDN_URL` in the [config](https://nuxt.com/docs/api/configuration/nuxt-config#cdnurl) as well, everything should work on its own without any extra setup.

**Reply by deeluxe74** [reply]
It's seems weird because the path is `images/banner` like localy, because the path is by default targe the public folder.
But i neither find a good explanation about dynamic image for nuxt3.

And like you said this not need extra setup, the best case is to find a easy solution that can be change when nuxt is update.
For me it was to put dynamic image inside public, for your case`CDN_URL` seems a good one !

**Reply by preetmishra** [reply]
Thanks for the help, @deeluxe74. I'll wait for a better solution too.

**Reply by sorcal** [reply]
I'm using nuxt 3.13.2 and I also have this issue. I put a computed value in img src and `app.cdnURL` is not automatically prepended to the image path in production build. I have to add the cdn url manually:
```vue
const photoSrc = computed(
  () => `$(app.cdnURL || ''}/images/${person === 'john' ? 'john-photo.webp' : 'default-photo.webp' }`,
)
```
